subiquitycore/controllers/network.py

```diff
@@ -294,10 +294,17 @@ def error(stage):
                 if os.path.exists('/lib/netplan/generate'):
                     # If netplan appears to be installed, run generate to
                     # at least test that what we wrote is acceptable to
-                    # netplan.
+                    # netplan but clear the SNAP environment variable to
+                    # avoid that netplan thinks its running in a snap and
+                    # tries to call netplan over the system bus.
+                    env = os.environ.copy()
+                    try:
+                        del env['SNAP']
```

Comment:
Minor: consier using the `contextlib.suppress` construct. It feels like verbose IMO.
```python
with contextlib.suppress(KeyError):
    del env['SNAP']